### PR TITLE
Fix issue with no-sourcemaps

### LINKED ISSUE #4390: Parcel 2 (latest nightly) generates a `.map` file with `--no-source-maps`.
# 🐛 bug report

Parcel 2 (latest nightly) generates a `.map` file with `--no-source-maps`.

To reproduce:

```
$ yarn parcel --version
2.0.0-nightly.180+a1a2515f

$ cat a.js
console.log(1 + 2);

$ rm -rf dist && yarn parcel build a.js && cat dist/a.js.map
dist/a.js    46 B     10ms
└── a.js     20 B    655ms

{"mappings":"AAAAA,QAAQC,IAAI","sources":["a.js"],"names":["console","log"],"version":3,"file":"a.js.map","sourcesContent":["console.log(1 + 2);\n"]}                                                                                          

$ rm -rf dist && y parcel build --no-source-maps a.js && cat dist/a.js.map
dist/a.js    46 B      2ms
└── a.js     20 B    655ms

{"mappings":"AACAA,QAAQC,IAAI","sources":["0"],"names":["console","log"],"version":3,"file":"a.js.map","sourcesContent":[null]}
```
With `--no-source-maps`, `dist/a.js.map` is still generated (it doesn't have `sourcesContent` though).

## 🤔 Expected Behavior

`.map` file should not be generated.

## 😯 Current Behavior

`.map` file is generated.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.0.0-nightly.180+a1a2515f
| Node             | v13.11.0
| npm/Yarn         | yarn 1.22.4
| Operating System | macOS 10.15.3

### PR DESCRIPTION
**Pull request by DeMoorJasper**
# ↪️ Pull Request

Fixes #4390

## 💻 Examples

Should not return a sourcemapReference when `sourcemaps` are disabled 

## ✔️ PR Todo

- [ ] Added/updated unit tests for this change
- [ ] Filled out test instructions (In case there aren't any unit tests)
- [ ] Included links to related issues/PRs

### CODE REVIEW

File: packages/optimizers/terser/src/TerserOptimizer.js [hunk 1]
```diff
@@ -58,7 +58,9 @@ export default new Optimizer({
         result.map.names || [],
       );
       let sourcemapReference: string = await getSourceMapReference(sourceMap);
-      minifiedContents += `\n//# sourceMappingURL=${sourcemapReference}\n`;
+      if (sourcemapReference) {
```
Comment (DeMoorJasper) [inline packages/optimizers/terser/src/TerserOptimizer.js]: optionally we could check `options.sourceMaps` instead?

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>packages/benchmarks/three 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

<details><summary>packages/benchmarks/kitchen-sink 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

<details><summary>packages/benchmarks/react-hn 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

<details><summary>packages/benchmarks/ak-editor 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/261167142815138305)**

**Comment by mischnic** [comment]
I think `getSourceMapReference` should rather return null when sourcemaps are disabled (it should also return null when there is no sourcemap right?).

**Comment by DeMoorJasper** [comment]
@mischnic couldn't figure out how to get flow to stop complaining so I solved it like this, will change it to null

**Comment by mischnic** [comment]
- I just noticed that we should disable set `sourceMap: false` in the terser config if they are disabled, might be faster. 
- The typing for terser are wrong, `result.map` has the type `empty` in the if statement (because the type is `?string | SourcemapObject`).

File: flow-libs/terser.js.flow [hunk 2]
```diff
@@ -144,15 +144,28 @@ declare module 'terser' {
     module?: boolean,
     output?: ?OutputOptions,
     parse?: ParseOptions,
-    nameCache?: ?{ [string]: string, ... },
+    nameCache?: ?{[string]: string, ...},
     safari10?: boolean,
     toplevel?: boolean,
     warnings?: boolean,
-    sourceMap?: SourceMapOptions
+    sourceMap?: SourceMapOptions,
```
Comment (mischnic) [inline flow-libs/terser.js.flow]:
```suggestion
    sourceMap?: false | SourceMapOptions,
```
and then you can remove the FlowFixMe 😉

Comment (DeMoorJasper) [reply flow-libs/terser.js.flow]: alright didn't know false was part of `?`

Comment (DeMoorJasper) [reply flow-libs/terser.js.flow]: Damn it I really need to disabled TS support in vscode for parcel...

**Pull request merged**